src/sfCommand.ts

```diff
@@ -445,6 +445,9 @@ export abstract class SfCommand<T> extends Command {
       code: codeFromError,
       // @ts-expect-error because context is not on Error
       context: (error.context as string) ?? null,
+      commandName: (error as SfCommand.Error).commandName ?? null,
```

Comment:
a useful pattern here uses `in` to narrow the type to one of the 3 in the union.

```
commandName: 'commandName' in error ? error.CommandName : null
```